src/transformers/models/wav2vec2/processing_wav2vec2_with_lm.py

```diff
@@ -0,0 +1,358 @@
+# coding=utf-8
+# Copyright 2021 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Speech processor class for Wav2Vec2
+"""
+import os
+from contextlib import contextmanager
+from dataclasses import dataclass
+from multiprocessing import Pool
+from typing import Iterable, List, Optional, Union
+
+import numpy as np
+
+from ...feature_extraction_utils import FeatureExtractionMixin
+from ...file_utils import ModelOutput, is_pyctcdecode_available, requires_backends
+from ...tokenization_utils import PreTrainedTokenizer
+from .feature_extraction_wav2vec2 import Wav2Vec2FeatureExtractor
+from .tokenization_wav2vec2 import Wav2Vec2CTCTokenizer
+
+
+if is_pyctcdecode_available():
+    from pyctcdecode import BeamSearchDecoderCTC
+    from pyctcdecode.alphabet import BLANK_TOKEN_PTN, UNK_TOKEN, UNK_TOKEN_PTN
+    from pyctcdecode.constants import (
+        DEFAULT_BEAM_WIDTH,
+        DEFAULT_HOTWORD_WEIGHT,
+        DEFAULT_MIN_TOKEN_LOGP,
+        DEFAULT_PRUNE_LOGP,
+    )
+
+
+@dataclass
+class Wav2Vec2DecoderWithLMOutput(ModelOutput):
+    """
+    Output type of :class:`~transformers.Wav2Vec2DecoderWith`, with transcription.
+
+    Args:
+        text (:obj:`list(str)`):
+            Decoded logits in text from. Usually the speech transcription.
+    """
+
+    text: Union[List[str], str]
+
+
+class Wav2Vec2ProcessorWithLM:
+    r"""
+    Constructs a Wav2Vec2 processor which wraps a Wav2Vec2 feature extractor, a Wav2Vec2 CTC tokenizer and a decoder
+    with language model support into a single processor for language model boosted speech recognition decoding.
+
+    Args:
+        feature_extractor (:obj:`Wav2Vec2FeatureExtractor`):
```

Comment:
Should use the class here for proper linking.
```suggestion
        feature_extractor (:class:`~transformers.Wav2Vec2FeatureExtractor`):
```